language/move-core/types/src/vm_status.rs

```diff
@@ -564,6 +564,7 @@ pub enum StatusCode {
     // The sender is trying to publish a module named `M`, but the sender's account already
     // contains a module with this name.
     DUPLICATE_MODULE_NAME = 1095,
+    CIRCULAR_DEPENDENCY = 1096,
```

Comment:
This should be an invariant violation error maybe? It's definitely not a verifier error thats for sure. 

The only way this can occur currently is with a writeset. Ideally long term, the writeset that causes this would be rejected. We might be able to do that today if we make this an invariant violation error (or something) and then try to load the module after the writeset is complete

Reply:
not sure about this. I hear you and the difficulties but don't know.
I think circular dependencies are definitely a verification step, whether that lives in the verifier directory is a different and orthogonal problem. But certainly something worth to think about.
The reason all of this seemed reasonable to me is because we keep talking and thinking about updating module via publishing and in that case we need this check and it has nothing to do with `WriteSet`.
But it's certainly true `WriteSet` is the only way to change code now...
so don't know.... I guess it's something to think about...